sdlib/d/gc/arena.d

```diff
@@ -751,3 +814,104 @@ unittest resizeLarge {
 	// Now full again:
 	assert(pd9.extent.hpd.full);
 }
+
+unittest resizeLargeGrow {
+	import d.gc.arena;
+	shared Arena arena;
+
+	auto base = &arena.base;
+	scope(exit) arena.base.clear();
+
+	import d.gc.emap;
+	static shared ExtentMap emap;
+	emap.tree.base = base;
+
+	import d.gc.region;
+	shared RegionAllocator regionAllocator;
+	regionAllocator.base = base;
+
+	arena.regionAllocator = &regionAllocator;
+
+	PageDescriptor makeAllocLarge(uint pages) {
+		auto ptr = arena.allocLarge(&emap, pages * PageSize);
+		assert(ptr !is null);
+		auto pd = emap.lookup(ptr);
+		assert(pd.extent.address is ptr);
+		assert(pd.extent.size == pages * PageSize);
+		return pd;
+	}
+
+	void checkGrowLarge(PageDescriptor pd, uint pages) {
```

Comment:
You'd be better off passing the extent down and reconstructing the proper page descriptor. The only thing you get from the one you pass in is the extent.